paasta_tools/tron_tools.py

```diff
@@ -415,21 +415,25 @@ def get_job_name(self):
     def get_action_name(self):
         return self.action
 
+    # mypy does not like the SecretVolume -> TronSecretVolume conversion, because TypedDict inheritence is broken. Until this is fixed, let's ignore this issue.
     def get_secret_volumes(self) -> List[TronSecretVolume]:  # type: ignore
         """Adds the secret_volume_name to the objet so tron/task_processing can load it downstream without replicating code."""
         secret_volumes = super().get_secret_volumes()
-        return [
-            TronSecretVolume(
+        tron_secret_volumes = []
+        for secret_volume in secret_volumes:
+            tron_secret_volume = TronSecretVolume(
                 secret_volume_name=self.get_secret_volume_name(
                     secret_volume["secret_name"]
                 ),
                 secret_name=secret_volume["secret_name"],
                 container_path=secret_volume["container_path"],
-                default_mode=secret_volume["default_mode"],
-                items=secret_volume["items"],
+                items=secret_volume.get("items", []),
             )
-            for secret_volume in secret_volumes
-        ]
+            if "default_mode" in secret_volume:
```

Comment:
would be nice to add a comment for why this is special-cased (we have different places where the default can come from and we don't want to insert the wrong default here)